### ISSUE TITLE
[markdown] add support for "file transclusion"

### ISSUE THREAD

**Issue by anb0s** [doxygen/doxygen#6845]
I'm using markdown files together with C++ code comments and PlantUML syntaxt.

Now i'm use embedded PlantUML files (e.g. component.puml) in markdown files. The syntaxt is from "MultiMarkdown": e.g. {{component.puml}}.

Then mdmerge (python script) is executed to merge (include) the files in to generated markdown file. This is done in script that calls doxygen after the preprocessing.

It would be good if doxygen can handle such "includes". May be i missing some information if it can be done in different way.

**Comment by albert-github** [comment]
When I understand it correctly the "File transclusion" is in fact a file as if it was written directly inline, like doxygen is doing with `\includedoc`.

**Comment by anb0s** [comment]
Yes, it's inlining of the file content, It's described here: http://fletcher.github.io/MultiMarkdown-5/transclusion.html

The tool i'm using for preprocessing is: https://pypi.org/project/MarkdownTools2/

P.S.: Markdown has lot of different flavors, so it is not consistent :( So i'm using only basic markdown syntaxt, but such a include feature is missing in markdown. AscciDoc would be a better solution, but not supported in doxygen.
